web/src/compose_recipient.js

```diff
@@ -108,25 +108,42 @@ export function update_on_recipient_change() {
     update_narrow_to_recipient_visibility();
 }
 
-export function check_stream_posting_policy_for_compose_box(stream_name) {
-    const stream = stream_data.get_sub_by_name(stream_name);
-    if (!stream) {
-        return;
+export function get_posting_policy_error_message() {
+    if (selected_recipient_id === "direct") {
+        if (
+            page_params.realm_private_message_policy ===
+            settings_config.private_message_policy_values.disabled.code
+        ) {
+            return $t({
+                defaultMessage: "Direct messages are disabled in this organization.",
+            });
+        }
+        return "";
     }
-    const can_post_messages_in_stream = stream_data.can_post_messages_in_stream(stream);
-    if (!can_post_messages_in_stream) {
-        $(".compose_right_float_container").addClass("disabled-compose-send-button-container");
-        compose_banner.show_error_message(
-            $t({
-                defaultMessage: "You do not have permission to post in this stream.",
-            }),
-            compose_banner.CLASSNAMES.no_post_permissions,
-            $("#compose_banners"),
-        );
-    } else {
+
+    const stream = stream_data.get_sub_by_id(selected_recipient_id);
+    if (stream && !stream_data.can_post_messages_in_stream(stream)) {
+        return $t({
+            defaultMessage: "You do not have permission to post in this stream.",
+        });
+    }
+    return "";
+}
+
+export function check_posting_policy_for_compose_box() {
+    const banner_text = get_posting_policy_error_message();
+    if (banner_text === "") {
         $(".compose_right_float_container").removeClass("disabled-compose-send-button-container");
         compose_banner.clear_errors();
+        return;
+    }
+
+    let banner_classname = compose_banner.CLASSNAMES.no_post_permissions;
```

Comment:
Now updated the class_name for banner also which was causing duplicate banner on sending , earlier we missed it.